src/ServiceControl/Infrastructure/Settings/DeprecatedConfigurationCheckFeature.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace ServiceBus.Management.Infrastructure.Settings
+{
+    using System;
+    using System.Threading.Tasks;
+    using System.Xml.Linq;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Logging;
+
+    class DeprecatedConfigurationCheckFeature : Feature
+    {
+        public DeprecatedConfigurationCheckFeature()
+        {
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context) => context.RegisterStartupTask(b => b.Build<DeprecatedConfigurationCheck>());
+
+        class DeprecatedConfigurationCheck : FeatureStartupTask
+        {
+            protected override Task OnStart(IMessageSession session)
+            {
+                var config = XDocument.Load(AppDomain.CurrentDomain.SetupInformation.ConfigurationFile);
```

Comment:
is there any chance for that code to throw an exception we need to catch?
e.g. can the configuration file path be null or empty?

Reply:
If there is no app.config file, ServiceControl won't start as there are config settings that are required. If all of the requried settings reside in the machine.config, then this code will return Task.CompletedTask and the user won't get warned about the config sections. [Docs for this property](https://msdn.microsoft.com/en-us/library/system.appdomainsetup.configurationfile.aspx)

If this code can't load the location specified by this property, it means the account that ServiceControl is running under doesn't have access to that disk location, in which case it shouldn't start anyway - at the very least, the codepath wouldn't actually get this far anyway.

So I don't know if there's any specific checks we need to do here, but I suspect not.